lib/matplotlib/stackplot.py

```diff
@@ -58,6 +58,15 @@ def stackplot(axes, x, *args,
     data : indexable object, optional
         DATA_PARAMETER_PLACEHOLDER
 
+    where: bool or array of bool, optional
```

Comment:
I'm struggling with the `where` parameter. While it is reasonable solve the issue internally by passing `where` to `fill_between`, is there any reasonable value for where other than `y!=0`?

- The generality of fill_between (where=) is not needed. It does not make sense to not plot regions in a stack plot where there are finite values. That would leave gaps.
- I assume `where` needs the shape of `y`; i.e. must be 2D. Not having a resolution in x does not make sense. Using the same where array for all datasets should be the very exception; usually you need.

It seems like a simpler high-level API such as a boolean parameter `clip_zero_regions` should be enough, and that would internally use `where=(y!=0)`.